odsclient/core.py

```diff
@@ -251,6 +256,32 @@ def get_whole_dataset(self,
 
         return result
 
+    # noinspection PyShadowingBuiltins
+    def push_dataset_realtime(self,
+                              dataset_id,         # type: str
+                              payload,            # type: List[Dict[str,Any]]
+                              push_key,           # type: str
+                              **other_opts
+                              ):
+        """
+        Pushes a Dataset.
+
+        :param dataset_id:
+        :param payload: The dataset to push as a list of dists, where the dict keys are the column names
```

Comment:
I am not very fond of  the name `payload`. Also from an API consistency point of view I think that we should consider providing either a dataframe or a csv string. So I would suggest to name this `dataset_csv_or_df` or just `dataset`, and to allow this to be either a pandas dataframe, a csv string, or a StringIO stream (please do not add this third option here and rather create another issue about it, since python2/3 compliance for streams is a bit tricky, I can handle it).

"list of dict" can be another allowed type (a 4th option) if you think that it makes sense ; in that case, and for consistency, we should open a new issue to try to allow the `get_` api to return it (separate from this PR)

Reply:
OK, Let's forget about list of dicts and keep CSV and Pandas as in the get functions. I've added these functionalities now.